src/inabox/inabox-iframe-messaging-client.js

```diff
@@ -51,13 +53,25 @@ function createIframeMessagingClient(win) {
   }
   iframeClient.setSentinel(sentinel || getRandom(win));
 
-  // Bet the top window is the scrollable window and loads host script.
-  // TODO(lannka,#9120):
-  // 1) check window ancestor origin, if the top window is in same origin,
-  // don't bother to use post messages.
-  // 2) broadcast the request
-  iframeClient.setHostWindow(win.top);
-  return iframeClient;
+  iframeClient.setBroadcastMode(true);
+  return new Promise(resolve => {
+    let hostWin = win;
+    let j = 0;
```

Comment:
these two vars would be clearer over by the loop

Reply:
Resolved by rewriting the loop.